internal/priorityqueue/priorityqueue.go

```diff
@@ -44,10 +44,42 @@ func (pq *PriorityQueue[T]) PopMax() T {
 	maxItem := pq.items[1]
 	pq.items[1] = pq.items[len(pq.items)-1]
 	pq.items = pq.items[:len(pq.items)-1]
-	pq.siftdown()
+	pq.siftdown(1)
 	return maxItem
 }
 
+// RemoveFunc removes all elements for which rm returns true.
+func (pq *PriorityQueue[T]) RemoveFunc(rm func(T) bool) {
+	// This is effectively slices.DeleteFunc, but inlined because we start from index 1,
+	// and we don't need to clear the tail.
```

Comment:
Why don't we need to clear the tail?

Reply:
There are no pointers in it, so it doesn't impact GC. (slices.DeleteFunc doesn't know one way or the other.) And nobody will ever read the tail. So clearing the tail is just wasted work.

Reply:
I must be missing something, but why do you say it has no pointers? T can be a pointer

Reply:
Oops. Done.